src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -783,8 +783,23 @@ public int GetENCBuildState(ShellInterop.ENC_BUILD_STATE[] pENCBuildState)
                     }
                     else if (_encService.EditSession != null)
                     {
-                        _projectBeingEmitted = _vsProject.VisualStudioWorkspace.CurrentSolution.GetProject(_vsProject.Id);
-                        _lastEditSessionSummary = GetProjectAnalysisSummary(_projectBeingEmitted);
+                        // Fetch the latest snapshot of the project and get an analysis summary for any changes 
+                        // made since the break mode was entered.
+                        var currentProject = _vsProject.VisualStudioWorkspace.CurrentSolution.GetProject(_vsProject.Id);
+                        if (currentProject == null)
+                        {
+                            // If the project has yet to be loaded into the solution (which may be the case,
+                            // since they are loaded on-demand), then it stands to reason that it has not yet
+                            // been modified.
+                            // TODO (https://github.com/dotnet/roslyn/issues/1204): this check should be unnecessary.
+                            _lastEditSessionSummary = ProjectAnalysisSummary.NoChanges;
```

Comment:
I think I saw at least an assert that _projectBeingEmitted wasn't null when _lastEditSessionSummary was examined elsewhere.